subprojects/logging/src/main/java/org/gradle/internal/logging/console/MultiLineBuildProgressArea.java

```diff
@@ -71,6 +63,23 @@ public int getHeight() {
     }
 
     @Override
+    public void resizeBuildProgressTo(int buildProgressLabelCount) {
+        int delta = buildProgressLabelCount - buildProgressLabels.size();
+        if (delta <= 0) {
+            // We don't support shrinking at the moment
+            return;
+        }
+
+        int row = parkingLabel.getWritePosition().row;
+        parkingLabel.scrollDownBy(delta);
+        while (delta-- > 0) {
+            DefaultRedrawableLabel label = newLabel(row--);
+            entries.add(entries.size() - 2, label);
```

Comment:
The `- 2` is because of the parking space and build summary, right? Seems like we could more strongly model this here to avoid errors.